pkg/plugin/clientmgmt/itemblockaction/v1/restartable_item_block_action.go

```diff
@@ -0,0 +1,115 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/plugin/clientmgmt/process"
+	"github.com/vmware-tanzu/velero/pkg/plugin/framework/common"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	ibav1 "github.com/vmware-tanzu/velero/pkg/plugin/velero/itemblockaction/v1"
+)
+
+// AdaptedItemBlockAction is a backup item action adapted to the v1 ItemBlockAction API
+type AdaptedItemBlockAction struct {
+	Kind common.PluginKind
+
+	// Get returns a restartable ItemBlockAction for the given name and process, wrapping if necessary
+	GetRestartable func(name string, restartableProcess process.RestartableProcess) ibav1.ItemBlockAction
+}
+
+func AdaptedItemBlockActions() []AdaptedItemBlockAction {
+	return []AdaptedItemBlockAction{
+		{
+			Kind: common.PluginKindItemBlockAction,
+			GetRestartable: func(name string, restartableProcess process.RestartableProcess) ibav1.ItemBlockAction {
+				return NewRestartableItemBlockAction(name, restartableProcess)
+			},
+		},
+	}
+}
+
+// RestartableItemBlockAction is an ItemBlock action for a given implementation (such as "pod"). It is associated with
+// a restartableProcess, which may be shared and used to run multiple plugins. At the beginning of each method
+// call, the restartableItemBlockAction asks its restartableProcess to restart itself if needed (e.g. if the
+// process terminated for any reason), then it proceeds with the actual call.
+type RestartableItemBlockAction struct {
+	Key                 process.KindAndName
+	SharedPluginProcess process.RestartableProcess
+}
+
+// NewRestartableItemBlockAction returns a new RestartableItemBlockAction.
+func NewRestartableItemBlockAction(name string, sharedPluginProcess process.RestartableProcess) *RestartableItemBlockAction {
+	r := &RestartableItemBlockAction{
+		Key:                 process.KindAndName{Kind: common.PluginKindItemBlockAction, Name: name},
+		SharedPluginProcess: sharedPluginProcess,
+	}
+	return r
+}
+
+// getItemBlockAction returns the ItemBlock action for this restartableItemBlockAction. It does *not* restart the
+// plugin process.
+func (r *RestartableItemBlockAction) getItemBlockAction() (ibav1.ItemBlockAction, error) {
+	plugin, err := r.SharedPluginProcess.GetByKindAndName(r.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	backupItemAction, ok := plugin.(ibav1.ItemBlockAction)
```

Comment:
backup item action is a good name here?

itemBlockAction would be better?

Reply:
@mateusoliveira43 oops. Copy-and-paste error. Will fix.